opendevin/memory/history.py

```diff
@@ -60,24 +67,29 @@ def get_events(self, reverse: bool = False) -> Iterable[Event]:
             if self.end_id != -1
             else self._event_stream.get_latest_event_id()
         )
-
+        summary_yielded = False
         for event in self._event_stream.get_events(
             start_id=start_id,
             end_id=end_id,
             reverse=reverse,
             filter_out_type=self.filter_out,
         ):
-            # TODO add summaries
-            # and filter out events that were included in a summary
-
             # filter out the events from a delegate of the current agent
-            if not any(
+            if (
+                self.summary is not None
+                and event.id <= self.last_summarized_event_id
+                and not summary_yielded
+            ):
+                summary_action = self.summary
+                summary_yielded = True
+                yield summary_action
```

Comment:
This doesn't get into the prompt, and I don't immediately see why. But it's not there, even in the case of: the first summarize action is done => agent continues and sends the regular prompt. This code *was* executed, but the prompt doesn't have the summary.